egs/wsj/s5/steps/nnet3/tdnn/train.sh

```diff
@@ -119,6 +123,15 @@ if [ $# != 4 ]; then
   echo "                                                   # Frame indices used for each splice layer."
   echo "                                                   # Format : layer<hidden_layer_index>/<frame_indices>....layer<hidden_layer>/<frame_indices> "
   echo "                                                   # (note: we splice processed, typically 40-dimensional frames"
+  echo "  --cnn-layer <string|cnn-layer0/filt_x_dim=3 filt_y_dim=8 filt_x_step=1 filt_y_step=1 num_filters=256 pool_x_size=1 pool_y_size=3 pool_z_size=1 pool_x_step=1 pool_y_step=3 pool_z_step=1> "
```

Comment:
Ideally we would call make_configs.py from the local/* scripts so this
would not be necessary in the future. I plan to take care of these things
when I write the steps/nnet3/train_dnn.py.

Tom basically wants to push his CNN recipe, with the current single layer
CNN configuration as he wants to focus on other things. The single layer
CNN recipe will not need to accept multiple versions of the same parameter,
so Tom may be you could simplify your life and not worry about providing
the multiple CNN layer configs to steps/nnet3/tdnn/train.sh. However ensure
that all other parts of the script can handle multiple CNN layers.

--Vijay

--VIjay

On Thu, Feb 25, 2016 at 3:19 PM, Daniel Povey <notifications@github.com>
wrote:

> In egs/wsj/s5/steps/nnet3/tdnn/train.sh
> <https://github.com/kaldi-asr/kaldi/pull/526#discussion_r54157653>:
>
> > @@ -119,6 +123,15 @@ if [ $# != 4 ]; then
> >    echo "                                                   # Frame indices used for each splice layer."
> >    echo "                                                   # Format : layer<hidden_layer_index>/<frame_indices>....layer<hidden_layer>/<frame_indices> "
> >    echo "                                                   # (note: we splice processed, typically 40-dimensional frames"
> > +  echo "  --cnn-layer <string|cnn-layer0/filt_x_dim=3 filt_y_dim=8 filt_x_step=1 filt_y_step=1 num_filters=256 pool_x_size=1 pool_y_size=3 pool_z_size=1 pool_x_step=1 pool_y_step=3 pool_z_step=1> "
>
> I'd rather not allow the shell script to accept multiple versions of the
> same parameter. This will be hard to implement and will break the normal
> shell-level interface. Dan
> … <#231389100_>
> On Thu, Feb 25, 2016 at 11:50 AM, Vijayaditya Peddinti <
> notifications@github.com> wrote: In egs/wsj/s5/steps/nnet3/tdnn/train.sh <#526
> (comment)
> <https://github.com/kaldi-asr/kaldi/pull/526#discussion_r54125545>>: > @@
> -119,6 +123,15 @@ if [ $# != 4 ]; then > echo " # Frame indices used for
> each splice layer." > echo " # Format :
> layer<hidden_layer_index>/<frame_indices>....layer<hidden_layer>/<frame_indices>
> " > echo " # (note: we splice processed, typically 40-dimensional frames" >
> + echo " --cnn-layer <string|cnn-layer0/filt_x_dim=3 filt_y_dim=8
> filt_x_step=1 filt_y_step=1 num_filters=256 pool_x_size=1 pool_y_size=3
> pool_z_size=1 pool_x_step=1 pool_y_step=3 pool_z_step=1> " does shell
> script allow for multiple specification of --cnn-layer parameters ? — Reply
> to this email directly or view it on GitHub <
> https://github.com/kaldi-asr/kaldi/pull/526/files#r54125545>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kaldi-asr/kaldi/pull/526/files#r54157653>.
>